acceptance-tests/src/test/java/io/blueocean/ath/live/GithubEditorTest.java

```diff
@@ -135,6 +135,28 @@ public void testEditor() throws IOException {
         sseClient.untilEvents(pipeline.buildsFinished);
     }
 
+    /**
+     * This test covers e2e usage of the editor, with a token that has whitespace added.
+     *
+     * Creates a blank github repo, and then uses editor to create a simple pipeline.
+     */
+    @Test
+    public void testEditorWithSpace() throws IOException {
+        creationPage.createPipeline(" " + token + " ", organization, repo, true);
+        MultiBranchPipeline pipeline = mbpFactory.pipeline(repo);
+        editorPage.simplePipeline();
+        ActivityPage activityPage = pipeline.getActivityPage().checkUrl();
+        driver.navigate().refresh();
+        sseClient.untilEvents(pipeline.buildsFinished);
+        sseClient.clear();
+        BranchPage branchPage = activityPage.clickBranchTab();
+        branchPage.openEditor("master");
+        editorPage.saveBranch("new - branch");
+        activityPage.checkUrl();
+        activityPage.getRunRowForBranch("new-branch");
+        sseClient.untilEvents(pipeline.buildsFinished);
+    }
```

Comment:
I think doing a full e2e creation + editor test just to validate the token trimming is overkill. Since the ATH can be flaky, I think we need to avoid creating overly long tests when we're really just testing one piece of functionality. I think we should add this as a test method on `GithubCreationTest`, the `testTokenValidation_failed` method would be good to emulate... we just need to feed in a token value padded with spaces and then assert that we moved on to the next step in the flow.

Reply:
Right on. `GithubCreationTest` has an `@After` method to delete the repo, and that will fire to do cleanup even if the test fails or errors out. Your test will create and save a valid token but I am pretty sure that `GithubCreationTest` and `GithubEditorTest` both have conditional logic in the underlying page object so that it can handle either a pristine creation flow w/ no token saved, or the flow when one is already saved. Arguably that's not the right way to do it but you shouldn't break anything. Famous last words :)